app/Http/Controllers/RequestWithdrawalController.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\RequestWithdrawal;
+use App\Http\Requests\StoreRequestWithdrawalRequest;
+use App\Http\Resources\RequestWithdrawalListingResource;
+use App\Http\Resources\RequestWithdrawalDetailedResource;
+use App\Models\RequestWithdrawalItem;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use App\Enums\OwnerType;
+use Throwable;
+
+class RequestWithdrawalController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        $withdrawals = RequestWithdrawal::with(['items.item', 'warehouse'])
+            ->latest()
+            ->paginate(config('app.pagination.per_page', 10));
+        return RequestWithdrawalListingResource::collection($withdrawals)
+            ->additional([
+                'message' => $withdrawals->isEmpty()?'No Request Withdrawals found.':'Request Withdrawals retrieved successfully.',
+                'success' => true,
+            ]);
+    }
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(StoreRequestWithdrawalRequest $request)
+    {
+        try {
+            $data = $request->validated();
+            $withdrawal = DB::transaction(function () use ($data) {
+                $withdrawal = RequestWithdrawal::create([
+                    'date_time'       => $data['date_time'],
+                    'warehouse_id'    => $data['warehouse_id'],
+                    'chargeable_id'   => $data['chargeable_id'],
+                    'chargeable_type' => OwnerType::from($data['chargeable_type'])->value,
+                    'equipment_no'    => $data['equipment_no'] ?? null,
+                    'smr'             => $data['smr'] ?? null,
+                    'fuel'            => $data['fuel'],
+                    'reference_no'    => $data['reference_no'] ?? null,
+                    'metadata'        => $data['metadata'] ?? null,
+                    'approvals'       => $data['approvals'] ?? [],
+                    'created_by'      => auth()->user()->id,
+                ]);
+                $items = collect($data['items'])->map(function ($item) use ($withdrawal) {
```

Comment:
based on mapped create from $data['items']
map is not necessary
you can directly create items through relationship
use $withdrawal->items->createMany($data['items']);